Upgrade deps to 2.0 and Generate icons from @phosphor/core

mrkpatchaa opened the pull request:
This will help in reducing number of files in this repo + upgrade to the most recent versions of icons.

mrkpatchaa commented:
@jhhayashi can you check this ?

jacesimons14 commented:
Would love to see this merged!!!!

mrkpatchaa commented:
In the meantime I published this npm package https://www.npmjs.com/package/@mrkpatchaa/phosphor-react-native with the latests updates.
I'll take id down once this PR is merged.

jacesimons14 commented:
sick thanks dude you're a king

jacesimons14 commented:
@mrkpatchaa I might just be an idiot, does your package require other dependencies? I'm seeing this
![image](https://github.com/duongdev/phosphor-react-native/assets/51728003/6a8fb46c-2c22-4cd8-bc47-32699fd9f07a)

mrkpatchaa commented:
Oh let me check. The typescript types are not pushed to npm.

mrkpatchaa commented:
@jacesimons14 can you check version 2.0.1? I fixed the types issue. 
Old version (this package)
<img width="750" alt="Screenshot 2023-08-10 at 12 20 52" src="https://github.com/duongdev/phosphor-react-native/assets/1266755/eb48844b-4d58-4f3f-a847-0b0257499c54">

Current version (my package)
<img width="735" alt="Screenshot 2023-08-10 at 12 20 58" src="https://github.com/duongdev/phosphor-react-native/assets/1266755/c0c1d96c-df64-4673-8efb-e6a765fd1830">

I also updated this PR with the fixes.

jacesimons14 commented:
@mrkpatchaa npm says there hasn't been a new version for 4 days

jacesimons14 commented:
still 2.0.0

mrkpatchaa commented:
@jacesimons14 maybe cache? This is the latest version on npm.
You can install the package again or run npm update to get the last version. 

<img width="439" alt="Screenshot 2023-08-11 at 10 03 49" src="https://github.com/duongdev/phosphor-react-native/assets/1266755/aceed5eb-d6b7-4235-8cb1-d0c57069f3d9">

jacesimons14 commented:
Ah I see it now, thank you, works great so far!

daxaxelrod commented:
Would love this. Running into issues where some icons are not available here but they are for the web package

jhhayashi commented:
> @jhhayashi can you check this ?

Hey there! I'm not affiliated with this repo, but happy to help where I can. 

This PR is a little too large to review. Would it be possible to split it into a few smaller ones?

mrkpatchaa commented:
@jhhayashi it's only "large" because of the deleted files that are not needed anymore. I am generating icons directly from phospor/core.

You can filter them out in the view. 
If you wish I can put them back and make the cleanup afterwards.

mrkpatchaa commented:
@jhhayashi I just pushed a new commit to revert the deletions. 
I will create another one for the deletion.

mrkpatchaa commented:
I just created the companion PR for this. #35

mrkpatchaa commented:
> Would love this. Have been running into issues where some icons are not available here but they are for the web package so this would be great

@daxaxelrod You can use the npm package I published as a temporary solution...
https://github.com/duongdev/phosphor-react-native/pull/33#issuecomment-1669682761

jhhayashi commented on generator/generate-svg.js:
```diff
@@ -50,22 +58,22 @@ const generateIconWithWeight = (icon, weight) => {
     tsCode = tsCode
       .replace('SvgProps, ', '')
-      .replace('function', "import { IconProps } from '../lib'\n\nfunction")
+      .replace('const', "import { IconProps } from '../lib'\n\nconst")
```
maybe better to just insert the import at the beginning of the string instead of hardcoding the first non-import line prefix?

jhhayashi commented on generator/generate-svg.js:
```diff
@@ -50,22 +58,22 @@ const generateIconWithWeight = (icon, weight) => {
     componentName: componentNameMap[componentName] || componentName,
   }).then((tsCode) => {
     tsCode = tsCode
       .replace('SvgProps, ', '')
```
this line can be removed (superseded by the `.replace(/import type .*;\n/g, '')` line)

jhhayashi commented on generator/generate-svg.js:
```diff
@@ -221,6 +229,28 @@ ${iconsExport}
+  const folders = [...Object.keys(weights), 'icons'];
+  for (let index = 0; index < folders.length; index++) {
+    try {
+      fs.rmSync(srcDir + '/' + folders[index], {
+        recursive: true,
+        force: true,
+      });
```
looks like https://github.com/jprichardson/node-fs-extra/blob/HEAD/docs/remove-sync.md can save you a few lines

jhhayashi reviewed (approved):
lgtm, but i am not affiliated with the repo

mrkpatchaa commented on generator/generate-svg.js:
```diff
@@ -36,11 +40,15 @@ const fileNameMap = {
+
+  const filePath = path.join(
+    svgsDir,
+    `${Case.capital(weight)}/${iconName}.svg`
```
We have control over the folder structure. In cases there are spaces We'd need to change more logic I think. Not only this line.

mrkpatchaa commented:
@duongdev @Joeao @kingdavidmartins @bkdev98 are you guys able to check this and merge? 
I created another npm package as a temporary solution but might be interesting to update this repo.

DavidP-B commented:
> In the meantime I published this npm package https://www.npmjs.com/package/@mrkpatchaa/phosphor-react-native with the latests updates. I'll take id down once this PR is merged.

I'm using your library meanwhile this PR is accepted. I've founded that you maybe forgot to import FileSearch icon... I'm using this icon but when I change to your library I can't see it.

mrkpatchaa commented:
@DavidP-B I just checked and file-search is an alias for file-magnifying-glass -> https://github.com/phosphor-icons/core/blob/main/bin/index.js
Maybe it would be interresting to add this as well in this PR. I'll try to find some time to do it this week. 
Anyway, seems like the repo is not maintained anymore. I couldn't get anyone to merge this 😕

duongdev commented:
Thank you @mrkpatchaa @DavidP-B and @jacesimons14 for your contribution. I am not currently work with React Native but your efforts on this library is so incredible!

mrkpatchaa commented:
> Thank you @mrkpatchaa @DavidP-B and @jacesimons14 for your contribution. I am not currently work with React Native but your efforts on this library is so incredible!

Hi @duongdev and happy to finally hear from you 😄 
What are the plans to continue maintaining the project? I can help if you want. 
In the meantime, with the new approach, it will require less work to maintain. 
Think also about the option of moving it to @phospor community?

jhhayashi commented:
Hey @mrkpatchaa!

@duongdev got me set up on this repo (see #34), so I'll now be able to help out with the maintenance. I have an active project that depends on this library, so I'll make sure to keep things maintained

mrkpatchaa commented:
Understood @jhhayashi . I also have an active project using it.
Feel free to let me know if you need help on something. 
Good luck mate.

DavidP-B commented:
@mrkpatchaa meanwhile there is no release of this PR, I'm using your library. It works fine in Expo Go and Android, but in iOS I got this error.

![image](https://github.com/duongdev/phosphor-react-native/assets/6517150/25fa4d8d-a3ed-43ab-ab1b-5e5fe1df7cea)

I'm missing something? I only change the repo in my package and all the paths to this new report.

jhhayashi commented:
Thanks for the patience; v2.0.0 has been released on npm

mrkpatchaa commented:
I will then remove @mrkpatchaa/phosphor-react-native from npm. 
Thanks.
